src/xunit.execution/Sdk/Reflection/Reflector.cs

```diff
@@ -58,6 +62,21 @@ internal static object ConvertArgument(object arg, Type type)
                     }
                     else
                     {
+                        if (type == typeof(Guid))
+                        {
+                            return Guid.Parse(arg.ToString());
+                        }
+
+                        if (type == typeof(DateTime))
+                        {
+                            return DateTime.Parse(arg.ToString(), CultureInfo.InvariantCulture);
```

Comment:
It would have to be stated in the docs the date time literal must be in an invariant culture compatible format.

Reply:
Hate to dredge up an old review @bradwilson, but does it make sense for the `arg`'s `toString` result to be used here? Shouldn't the parsing only happen if `arg` is actually a `string`?